intermol/lammps/lammps_parser.py

```diff
@@ -198,12 +200,12 @@ def canonical_dihedral(self, params, dihedral, direction='into'):
                         typename = 'charmm'
                         paramlist = convert_dihedral_from_trig_to_proper(params)
 
-            elif dihedral ==  ImproperHarmonicDihedral:
+            elif dihedral == ImproperHarmonicDihedral:
                 params['k'] *= canonical_force_scale
                 typename = 'harmonic'
                 paramlist = [params]
             else:
-                raise ValueError('A non-canonical dihedral was found in the '
+                raise InterMolError('A non-canonical dihedral was found in the '
                                  'system. All dihedrals should have been '
                                  'converted to either TrigDihedralType or '
                                  'ImproperHarmonicType so something likely '
```

Comment:
I changed these a bit in my commit.  Will have to merge later, but I think it's more general.